src/App.tsx

```diff
@@ -1,61 +1,136 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+import { FormEvent, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
+
+function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+export const todosWithUsers = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(todosWithUsers);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [hasError, setHasError] = useState({
+    title: false,
+    userId: false,
+  });
+
+  const maxId = Math.max(...todos.map(todo => todo.id)) + 1;
+
+  const handleTitle = (event: FormEvent) => {
+    const target = event.target as HTMLInputElement;
+
+    setTitle(target.value);
+    setHasError(prev => ({
+      ...prev,
+      title: false,
+    }));
+  };
+
+  const handleUserId = (event: FormEvent) => {
+    const target = event.target as HTMLSelectElement;
+
+    setUserId(+target.value);
+    setHasError(prev => ({
+      ...prev,
+      userId: false,
+    }));
+  };
+
+  const addNewTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title) {
+      setHasError(prev => ({
+        ...prev,
+        title: true,
+      }));
+    }
+
+    if (!userId) {
+      setHasError(prev => ({
+        ...prev,
+        userId: true,
+      }));
+    }
+
+    if (!title || !userId) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: maxId,
+      title,
+      completed: false,
+      userId,
+      user: getUserById(userId),
+    };
+
+    setTodos(prev => [...prev, newTodo]);
+    setTitle('');
+    setUserId(0);
+  };
```

Comment:
[CODE KNOWLEDGE] - While handling form values, trigger submit logic using `onSubmit` prop of form, instead of `onClick` submit button. You are correctly using `onSubmit` prop of form to handle form submission. This is a good practice as it allows the form to handle submission even when the submit button is not clicked, for example, when the user presses the Enter key.